Include/internal/pycore_object.h

```diff
@@ -279,7 +279,7 @@ static inline size_t
 _PyType_PreHeaderSize(PyTypeObject *tp)
 {
     return _PyType_IS_GC(tp) * sizeof(PyGC_Head) +
-        _PyType_HasFeature(tp, Py_TPFLAGS_MANAGED_DICT) * 2 * sizeof(PyObject *);
+        _PyType_HasFeature(tp, Py_TPFLAGS_PREHEADER) * 2 * sizeof(PyObject *);
```

Comment:
I would have expected the size to change.  Were we over-allocating before?

Reply:
Yes, and no.

We were over-allocating in terms of space, but we need to allocate in 2 words blocks because of alignment requirements.
C requires that `malloc` aligns to the largest thing, i.e. `long double`.